scripts/partner_batch_imports.py

```diff
@@ -0,0 +1,197 @@
+        ]
+
+        # Inactive fields
+        self.copyright = data[19]
+        self.issn = data[54]
+        self.doi = data[145]
+        self.lccn = data[146]
+        self.dewey = data[49]
+        # physical_dimensions
+        # e.g. "5.4 x 4.7 x 0.2 inches"
+        self.length, self.width, self.height = data[40:43]
+
+        # Assert importable
+        assert self.isbn_13
+        for field in self.REQUIRED_FIELDS:
+            assert getattr(self, field)
+
+    @staticmethod
+    def contributors(data):
+        def make_author(name, _, typ):
+            author = {'name': name}
+            if typ == 'X':
+                # set corporate contributor
+                author['entity_type'] = 'org'
+            # TODO: sort out contributor types
+            # AU = author
+            # ED = editor
+            return author
+
+        contributors = (
+            (data[21+i*3], data[22+i*3], data[23+i*3]) for i in range(5)
+        )
+
+        # form list of author dicts
+        authors = [make_author(*c) for c in contributors if c[0]]
+        return authors
+
+    def json(self):
+        return {
+            field: getattr(self, field)
+            for field in self.ACTIVE_FIELDS
+            if getattr(self, field)
+        }
+
+
+def load_state(path, logfile):
+    """Retrieves starting point from logfile, if log exists
+
+    Takes as input a path which expands to an ordered candidate list
+    of bettworldbks* filenames to process, the location of the
+    logfile, and determines which of those files are remaining, as
+    well as what our offset is in that file.
+
+    e.g. if we request path containing f1, f2, f3 and our log
+    says f2,100 then we start our processing at f2 at the 100th line.
+
+    This assumes the script is being called w/ e.g.:
+    /1/var/tmp/imports/2021-08/Bibliographic/*/
+    """
+    filenames = sorted([
+        os.path.join(path, f)
+        for f in os.listdir(path)
+        if f.startswith("bettworldbks")
+    ])
+    try:
+        with open(logfile) as fin:
+            active_fname, offset = next(fin).strip().split(',')
+            unfinished_filenames = filenames[filenames.index(active_fname):]
+            return unfinished_filenames, int(offset)
+    except (ValueError, OSError):
+        return filenames, 0
+
+
+def update_state(logfile, fname, line_num=0):
+    """Records the last file we began processing and the current line"""
+    with open(logfile, 'w') as fout:
+        fout.write('%s,%s\n' % (fname, line_num))
+
+
+def csv_to_ol_json_item(line):
+    """converts a line to a book item"""
+    b = Biblio(line.strip().split('|'))
+    return {
+        'ia_id': 'isbn:%s' % b.isbn,
+        'data': b.json()
+    }
+
+
+def batch_import(path, batch, batch_size=5000):
+    logfile = os.path.join(path, 'import.log')
+    filenames, offset = load_state(path, logfile)
+
+    for fname in filenames:
+        book_items = []
+        with open(fname, 'r', encoding="ISO-8859-1") as f:
```

Comment:
Are you sure the input file is meant to be ISO-8859-1 encoded?
I'm not sure exactly what file you are using, but I though all partner input files claimed to be UTF-8 encoded.

If there are  ISO-8859-1 characters in there, I'd be concerned those are errors, and assuming the entire file is ISO-8859-1 encoded will break the correctly encoded UTF-8.

Reply:
```suggestion
        with open(fname, 'r') as f:
```